new-log-viewer/src/components/StatusBar/LogLevelSelect/index.tsx

```diff
@@ -0,0 +1,273 @@
+ * @param props.children
+ * @param props.rest
+ * @return
+ */
+const PlacementLeftTooltip = ({children, ...rest}: TooltipProps) => (
+    <Tooltip
+        {...rest}
+        placement={"left"}
+    >
+        {children}
+    </Tooltip>
+);
+
+/**
+ * Renders a `<Tooltip/>` with its placement set to the right of the children element.
+ *
+ * @param props
+ * @param props.children
+ * @param props.rest
+ * @return
+ */
+const PlacementRightTooltip = ({children, ...rest}: TooltipProps) => (
+    <Tooltip
+        {...rest}
+        placement={"right"}
+    >
+        {children}
+    </Tooltip>
+);
+
+interface LogSelectOptionProps {
+    isChecked: boolean,
+    logLevelName: string,
+    logLevelValue: LOG_LEVEL,
+    onCheckboxClick: React.MouseEventHandler
+}
+
+/**
+ * Renders an <Option/> in the <LogLevelSelect/> for selecting some log level and the level above
+ * it.
+ *
+ * @param props
+ * @param props.isChecked
+ * @param props.logLevelName
+ * @param props.logLevelValue
+ * @param props.onCheckboxClick
+ * @return
+ */
+const LogSelectOption = ({
+    isChecked,
+    logLevelName,
+    logLevelValue,
+    onCheckboxClick,
+}:LogSelectOptionProps) => {
+    return (
+        <Option
+            data-value={logLevelValue}
+            key={logLevelName}
+            value={logLevelValue}
+        >
+            <ListItemDecorator>
+                <PlacementLeftTooltip
+                    title={
+                        <Stack
+                            alignItems={"center"}
+                            direction={"row"}
+                        >
+                            {isChecked ?
+                                <RemoveIcon/> :
+                                <AddIcon/>}
+                            {logLevelName}
+                        </Stack>
+                    }
+                >
+                    <Checkbox
+                        checked={isChecked}
+                        size={"sm"}
+                        value={logLevelValue}
+                        onClick={onCheckboxClick}/>
+                </PlacementLeftTooltip>
+            </ListItemDecorator>
+            <PlacementRightTooltip
+                title={
+                    <Stack
+                        alignItems={"center"}
+                        direction={"row"}
+                    >
+                        {logLevelName}
+                        {" and below"}
+                    </Stack>
+                }
+            >
+                <ListItemContent data-value={logLevelValue}>
+                    {logLevelName}
+                </ListItemContent>
+            </PlacementRightTooltip>
+        </Option>
+    );
+};
+
+interface ClearFiltersOptionProps {
+    onClick: () => void
+}
+
+/**
+ * Renders an <Option/> to clear all filters in the <LogLevelSelect/>.
+ *
+ * @param props
+ * @param props.onClick
+ * @return
+ */
+const ClearFiltersOption = ({onClick}: ClearFiltersOptionProps) => (
+    <Option
+        value={-1}
```

Comment:
I think the current behaviour for this option's code is a bit unintuitive. For the normal select options, we use data-value to determine the start of the range. But for this one, data-value is unset, so the start of the range ends up being NaN. In that case, perhaps we should just check whether data-value is set and if not, we clear the range? And in that case, it doesn't really matter what we set for value here.

Reply:
sg

Reply:
In fact, now I recall why I supplied this `-1` value - it's a mandatory attribute of `<Option/>`. Since we need the `<Option/>` here only because we want a button that appears the same as the other options, the value is unimportant here.

`null` works. Shall we keep it as `null`?

Reply:
Sg